tools/deploy.py

```diff
@@ -260,22 +257,27 @@ def generate_manifest(args):
     manifest_data = collect_manifest_data(job_data)
     print("Manifest:", json.dumps(manifest_data, indent=2, sort_keys=True))
 
-    # Write the manifest.json for subsequent deployment to the server
-    with open(output_destpath, mode="w") as fp:
-        json.dump(manifest_data, fp, indent=2, sort_keys=True)
-
     # If possible, check if the remote manifest is the same as our local one
-    remote_manifest_data = None
+    remote_manifest_data = {}
     if args.dest_url:
         # Check if generated manifest.json file differs from the one that
         # is currently deployed.
         manifest_url = f"{args.dest_url}/{destpath}"
         manifest_url = manifest_url.format_map(format_data)
 
         try:
-            remote_manifest_data = url_fetch(manifest_url)
-        except IOError:
-            pass
+            remote_manifest_data = url_download_json(manifest_url)
+        except IOError as e:
+            print("Fetching remote manifest failed: %s", e.msg)
+
+    if args.update:
+        for key, value in remote_manifest_data.items():
+            if key not in manifest_data:
+                manifest_data[key] = value
```

Comment:
this assumes the remote_manifest contains data, can we reasonably expect this all the time? Why did you decide against `remote_manifest_data.update()`?

Reply:
If it doesn't it's an empty dict so that the entire `for` loop is a no-op.

Reply:
Ah right, misinterpreted a previous error. Of course an empty loop will do nothing.